src/gui/editorwidgets/qgsrelationreferenceconfigdlg.cpp

```diff
@@ -47,7 +47,7 @@ QgsRelationReferenceConfigDlg::QgsRelationReferenceConfigDlg( QgsVectorLayer* vl
 
   Q_FOREACH ( const QgsRelation& relation, vl->referencingRelations( fieldIdx ) )
   {
-    mComboRelation->addItem( QString( "%1 (%2)" ).arg( relation.id(), relation.referencedLayerId() ), relation.id() );
+    mComboRelation->addItem( QString( "%1 (%2)" ).arg( relation.name(), relation.referencedLayerId() ), relation.id() );
```

Comment:
Names can be left empty, I'd say we either need to enforce that a name is never empty or fallback to the id in case the name is empty.

Reply:
How about simplifying the auto generated IDs too? Trimming the unique id to a couple of characters should be ok

Reply:
Thanks guys. I prefer the fallback than the "never empty" because of the backwards compatibility with older projects.
Simplifying the ids is a nice idea too. I'll check the possibilities. Do you mean in general or just here to display in the combo?

Reply:
In general

Reply:
The relationid consists of the field names and the ids of the referencing and the referenced layer. These layer ids could be quite long. So another question appears - should the layer ids be simplified  in general or should we simplify them only as component of the relationid?
(the relationid is anyway unique because of a suffix function)

Reply:
To keep this pull request in a state where it can be merged without much discussion, I would avoid changing the ids themselves.
The other question would need a wider discussion (but sounds like a good plan)